src/reader.rs

```diff
@@ -1763,4 +1758,27 @@ mod tests {
             Kml::KmlDocument(_)
         ));
     }
+
+    #[test]
+    fn test_parse_empty_coordinates() {
+        let kml_str = r#"
+            <?xml version="1.0" encoding="utf-8"?>
+            <kml xmlns="http://www.opengis.net/kml/2.2">
+              <Document>
+                <Placemark>
+                  <name>test</name>
+                  <LineString>
+                    <altitudeMode>absolute</altitudeMode>
+                    <coordinates />
+                  </LineString>
+                </Placemark>
+              </Document>
+            </kml>
+        "#;
+
+        assert!(matches!(
+            Kml::<f64>::from_str(kml_str).unwrap(),
+            Kml::KmlDocument(_)
+        ));
```

Comment:
It looks like this current test isn't asserting that the coordinates are an empty vec. Something like this should work though and verify that it passes successfully

```suggestion
        let kml_str = r#"
        <LineString>
            <altitudeMode>absolute</altitudeMode>
            <coordinates />
        </LineString>
        "#;

        assert_eq!(
            Kml::<f64>::from_str(kml_str).unwrap(),
            Kml::LineString(LineString {
                coords: vec![],
                altitude_mode: types::AltitudeMode::Absolute,
                ..Default::default()
            })
        );
```